src/rust/engine/src/nodes.rs

```diff
@@ -1049,6 +1063,22 @@ impl Node for NodeKey {
       NodeKey::Snapshot(n) => n.run(context).map(|v| v.into()).to_boxed(),
       NodeKey::Task(n) => n.run(context).map(|v| v.into()).to_boxed(),
     }
+    .inspect(move |_: &NodeResult| {
+      if context2.session.should_record_zipkin_spans() {
```

Comment:
With this flag, I want to say explicitly that will run the code only if spans for v2 engine are collected. Also, I think that the flag `should_record_zipkin_spans` is a temporary one and exists only because we need to add code that decides for which Nodes we want to collect Zipkin spans.

Reply:
I agree with @cosmicexplorer that having this check one time above to set an `Option` to either `Some` or `None`, and here just checking if that value is `Some` or `None` is probably a little cleaner than doing the same check twice :) It means that if we change how we do the first deciding, we don't need to change this second piece of code too.